src/promises/allNoLeaks.ts

```diff
@@ -0,0 +1,38 @@
+// From https://stackoverflow.com/questions/64928212/how-to-use-promise-allsettled-with-typescript#69451755
+const isRejected = (
+	input: PromiseSettledResult<unknown>,
+): input is PromiseRejectedResult => input.status === 'rejected';
+
+// From https://stackoverflow.com/questions/64928212/how-to-use-promise-allsettled-with-typescript#69451755
+const isFulfilled = <T>(
+	input: PromiseSettledResult<T>,
+): input is PromiseFulfilledResult<T> => input.status === 'fulfilled';
+
+// The contract of PromiseRejectedResult requires `reason` to be `any`. Thus lint is disabled here.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getErrorFromReason = (reason: any): Error => {
```

Comment:
We shouldn't use any -- that isn't safe -- but rather use `unknown` (which still means it could be anything, but if you try to access an attribute without a typeguard then typescript will scream at you)

Changing to `unknown` will let you remove the eslint override too!

Reply:
Hmm, IIRC `unknown` might have come with its own trouble, but I'll try it again and see.